connect/transforms/src/main/java/org/apache/kafka/connect/transforms/field/SingleFieldPath.java

```diff
@@ -202,6 +208,161 @@ public Object valueFrom(Map<String, Object> map) {
         return current.get(lastStep());
     }
 
+    /**
+     * Updates the matching field value if found
+     * @param map original root value
+     * @param update function to apply to existing value
+     * @return the original value with the matching field updated if found
+     */
+    public Map<String, Object> updateMap(
+        Map<String, Object> map,
+        Function<Object, Object> update
+    ) {
+        if (map == null) return null;
+        Map<String, Object> result = new HashMap<>(map);
+
+        Map<String, Object> parent = result;
+        Map<String, Object> child;
+        for (String step : stepsWithoutLast()) {
+            child = requireMapOrNull(parent.get(step), "nested field access");
+            if (child == null) return map;
+            child = new HashMap<>(child);
+            parent.put(step, child);
+            parent = child;
+        }
+
+        Object original = parent.get(lastStep());
+        Object updated = update.apply(original);
+        if (updated != null) {
+            parent.put(lastStep(), updated);
+        }
```

Comment:
It's a little strange to invoke the update function even if `original` is null. Maybe something like this would be more intuitive?

```suggestion
        if (original == null)
            return map;

        Object updated = update.apply(original);
        if (updated != null) {
            parent.put(lastStep(), updated);
        }
```

Or, if we _really_ want to mirror the exact current behavior of `TimestampConverter` (where `null` is inserted when the field isn't found), we could do something like this:
```suggestion
        Object updated = original != null ? update.apply(original) : null;
        parent.put(lastStep(), updated);
```